python/pyarrow/tests/test_dataset.py

```diff
@@ -1802,6 +1802,29 @@ def test_open_dataset_from_fsspec(tempdir):
     assert dataset.schema.equals(table.schema)
 
 
+def test_filter_timestamp(tempdir):
+    # ARROW-11379
+    import pyarrow.parquet as pq
+    path = tempdir / "test_partition_timestamps"
+
+    table = pa.table({
+        "dates": ['2012-01-01', '2012-01-02'] * 5,
+        "id": range(10)})
+
+    # write dataset partitioned on dates (as strings)
+    part = ds.partitioning(table.select(['dates']).schema, flavor="hive")
+    ds.write_dataset(table, path, partitioning=part, format="feather")
+
+    # read dataset partitioned on dates (as timestamps)
+    part = ds.partitioning(pa.schema([("dates", pa.timestamp("s"))]),
+                           flavor="hive")
+    dataset = ds.dataset(path, format="feather", partitioning=part)
+
+    condition = ds.field("dates") > pd.Timestamp("2012-01-01")
```

Comment:
this line still causes a pandas dependency 
It could be avoided by using `datetime.datetime`, but I think it is good to test both, because pandas Timestamp has nanosecond resolution, it gives a different casting need)

Reply:
Something else: does filtering with a string still work (and the string being interpreted as timestamp): `ds.field("dates") > "2012-01-01"` (it did work based on my comment in the issue, so would be good to add here).

(and if we have three values to use in the filter, maybe put it in a for loop or pytest parametrize to avoid repeating the rest of the code)